src/main/java/org/broadinstitute/hellbender/tools/spark/sv/SVIntervalTree.java

```diff
@@ -0,0 +1,904 @@
+
+                    node = node.right;
+                }
+            }
+
+            return 0;
+        }
+
+        private int getRank() {
+            int result = 1;
+            if ( left != null )
+                result = left.size + 1;
+            return result;
+        }
+
+        private Node<V1> spliceOut( final Node<V1> child, final Node<V1> initialRoot ) {
+            Node<V1> root = initialRoot;
+            if ( (child.parent = parent) == null ) {
+                root = child;
+                child.isBlack = true;
+            } else {
+                if ( parent.left == this )
+                    parent.left = child;
+                else
+                    parent.right = child;
+                fixup(parent);
+
+                if ( isBlack )
+                    root = removeFixup(parent, child, root);
+            }
+
+            return root;
+        }
+
+        private Node<V1> rotateLeft( final Node<V1> initialRoot ) {
+            Node<V1> root = initialRoot;
+            final Node<V1> child = right;
+
+            final int childSize = child.size;
+            child.size = size;
+            size -= childSize;
+
+            if ( (right = child.left) != null ) {
+                right.parent = this;
+                size += right.size;
+            }
+
+            if ( (child.parent = parent) == null )
+                root = child;
+            else if ( this == parent.left )
+                parent.left = child;
+            else
+                parent.right = child;
+
+            child.left = this;
+            parent = child;
+
+            setMaxEnd();
+            child.setMaxEnd();
+
+            return root;
+        }
+
+        private Node<V1> rotateRight( final Node<V1> initialRoot ) {
+            Node<V1> root = initialRoot;
+            final Node<V1> child = left;
+
+            final int childSize = child.size;
+            child.size = size;
+            size -= childSize;
+
+            if ( (left = child.right) != null ) {
+                left.parent = this;
+                size += left.size;
+            }
+
+            if ( (child.parent = parent) == null )
+                root = child;
+            else if ( this == parent.left )
+                parent.left = child;
+            else
+                parent.right = child;
+
+            child.right = this;
+            parent = child;
+
+            setMaxEnd();
+            child.setMaxEnd();
+
+            return root;
+        }
+
+        private void setMaxEnd() {
+            maxEndInterval = interval;
+            if ( left != null ) maxEndInterval = laterEndingInterval(maxEndInterval, left.maxEndInterval);
+            if ( right != null ) maxEndInterval = laterEndingInterval(maxEndInterval, right.maxEndInterval);
+        }
+
+        private static SVInterval laterEndingInterval( final SVInterval interval1, final SVInterval interval2 ) {
+            final int contig1 = interval1.getContig();
```

Comment:
Unfortunately, no.  The natural order comparison is implemented on {contig, start, end} in that order.  Here I'm implementing the sort order {contig, end}.